### PR TITLE
Find Smallest Letter Greater Than Target | issue #17

### PR DESCRIPTION
**Pull request by namita0210**
- Written the code in java in "others folder"
- Updated question.md --> added problem statement, solution steps , name of java file that consists of the code.
 
image

Please merge my PR :)

### CODE REVIEW

File: Others/Name Of Question (template)/q1.java
Comment (bh-g) [inline Others/Name Of Question (template)/q1.java:1]: Make a new Folder in Others with the name: Find Smallest Letter Greater Than Target.
Inside that make two new files question.md and solution.java and add your question and solution there.

**Review by bh-g — CHANGES_REQUESTED** [review]

File: Others/Name Of Question (template)/q1.java
Comment (namita0210) [reply Others/Name Of Question (template)/q1.java:1]: - Made the changes in the  forked repo
- committed the changes
**Question :** PR was already created -  I made the suggested changes now . Do I need to create PR again?

File: Others/Name Of Question (template)/Find Smallest Letter Greater Than Target/question.md
Comment (bh-g) [inline Others/Name Of Question (template)/Find Smallest Letter Greater Than Target/question.md:1]: Move the folder named Find Smallest Letter Greater Than Target inside Others and not inside the template, look at word search for reference.

File: Others/Name Of Question (template)/Find Smallest Letter Greater Than Target/question.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+###Write the question here!
```
Comment (bh-g) [inline Others/Name Of Question (template)/Find Smallest Letter Greater Than Target/question.md:1]: Remove this line.

File: Others/Name Of Question (template)/Find Smallest Letter Greater Than Target/question.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+<p>
+  <b color="blue"> Q1. Find smallest letter greater than target</b><br>
```
Comment (bh-g) [inline Others/Name Of Question (template)/Find Smallest Letter Greater Than Target/question.md:3]: Remove 'Q1'.

File: Others/Name Of Question (template)/Find Smallest Letter Greater Than Target/question.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+Return the smallest character in letters that is lexicographically greater than target. If such a character does not exist, return the first character in letters.</p>
+
+<p> Solution Steps <li>Parse array</li>
+<li>If target> arr[i] --> i++</li>
+<li>If target is not > arr[i] --> return arr[i]</li>
+<li> If even at the end of parsing the array there is no character that fulfills the if condition --> return arr[0]</li></p>
+
+<u>Solution in file : q1.java</u>
+
+
```
Comment (bh-g) [inline Others/Name Of Question (template)/Find Smallest Letter Greater Than Target/question.md:14]: Delete this.

File: Others/Name Of Question (template)/Find Smallest Letter Greater Than Target/question.md
Comment (bh-g) [inline Others/Name Of Question (template)/Find Smallest Letter Greater Than Target/question.md:1]: Add examples in the question.

File: Others/Name Of Question (template)/Find Smallest Letter Greater Than Target/solution.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
+public class solution{
+
```
Comment (bh-g) [inline Others/Name Of Question (template)/Find Smallest Letter Greater Than Target/solution.java:2]: Delete this.

File: Others/Name Of Question (template)/Find Smallest Letter Greater Than Target/solution.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+    }
+
+    public static void main(String[]args){
+        char[] letters = {'c','f','j'};
+        char target = 'c';
+        char ans ;
+        ans =nextGreatestLetter(letters,target);
+        System.out.println(ans);
+    }
+}
```
Comment (bh-g) [inline Others/Name Of Question (template)/Find Smallest Letter Greater Than Target/solution.java:19]: Not needed, delete this.

File: Others/Name Of Question (template)/Find Smallest Letter Greater Than Target/solution.java [hunk 6]
```diff
@@ -0,0 +1,19 @@
+public class solution{
+
+    public static char nextGreatestLetter(char[] letters, char target) {
+        for (int i = 0 ; i < letters.length ; i++){
+            if(target<letters[i]){
+                return letters[i];
+            }
+        }
+        return letters[0];
+    }
```
Comment (bh-g) [inline Others/Name Of Question (template)/Find Smallest Letter Greater Than Target/solution.java:10]: Re-indent this part after removing rest of the code.

**Review by bh-g — CHANGES_REQUESTED** [review]

**Comment by namita0210** [comment]
### Issue #17  Fixed

- Solution snapshot

- image

- Question snapshot

- image

**Comment by namita0210** [comment]
> ### Issue #17 Fixed
> * Solution snapshot
> * image
> * Question snapshot
> * image

Hi @bh-g  All the changes have been made - Can you please merge this PR ?

**Pull request merged**